packages/Popover/src/Popover.spec.js

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import { render, cleanup } from 'react-testing-library';
+import Popover, { Align, Justify } from './Popover';
+import { calculatePosition, getElementPosition } from './positionUtils';
+
+afterAll(cleanup);
+
+describe('packages/Popover', () => {
+  const { container, unmount } = render(
+    <>
+      <button>Trigger Element</button>
+      <Popover justify={Justify.Start} align={Align.Top} active>
+        Content to appear inside of Popover component
+      </Popover>
+    </>,
+  );
+
+  test('should show popover when trigger is clicked', () => {
+    // eslint-disbale
+    expect(document.body.children[1].firstChild.innerHTML).toBe(
+      'Content to appear inside of Popover component',
+    );
+  });
+
+  test('renders children inside of a portaled component', () => {
+    expect(
+      container.innerHTML.includes('<button>Trigger Element</button>'),
+    ).toBe(true);
+
+    expect(
+      container.innerHTML.includes(
+        'Content to appear inside of Popover component',
+      ),
+    ).toBe(false);
+  });
+
+  test('does not render children when active is false', () => {
+    const wrapper = render(
+      <>
+        <button>Trigger Element</button>
+        <Popover justify={Justify.Start} align={Align.Top}>
+          Content to appear inside of Popover component
+        </Popover>
+      </>,
+    );
+
+    expect(
+      wrapper.container.innerHTML.includes(
+        'Content to appear inside of Popover component',
+      ),
+    ).toBe(false);
+  });
+
+  test('does not Portal Popover component, when usePortal is false', () => {
+    const { container } = render(
+      <>
+        <button>Trigger Element</button>
+        <Popover
+          active
+          usePortal={false}
+          justify={Justify.Start}
+          align={Align.Top}
+        >
+          Popover in DOM
+        </Popover>
+      </>,
+    );
+
+    expect(container.innerHTML.includes('Popover in DOM')).toBe(true);
+  });
+
+  test('removes Popover instance on unmount', () => {
+    unmount();
+    expect(container.innerHTML).toBe('');
+  });
+
+  describe('unit tests', () => {
```

Comment:
I'd even make a new file positionUtils.spec.js so we can have them all consolidated.

We should have tests that:

- Top, left, bottom, right viewport collisions. Both when a position is within that boundary and conflicting with it. Such that if we have a popover that's aligned at Top we want to have a test that tests when Top is fine, when we have to move to Align.Bottom, and then to Align.Middle.
  - Need to test absolute and relative positioning for every permutation.

The goal being that we have courage that should we decide to refactor the code we can do so without worry of any regressions, so we need to write tests for any edge cases.

Reply:
Makes sense, will start working on it 👍